spec/requests/api/v1/inventory_list_items_controller_spec.rb

```diff
@@ -0,0 +1,351 @@
+      invalid_request_body[:data].delete(:type)
+      hash_body = nil
+      post api_v1_team_inventory_column_list_items_path(
+        team_id: @teams.first.id,
+        inventory_id: @valid_inventory.id,
+        column_id: @list_column
+      ), params: invalid_request_body.to_json, headers: @valid_headers
+      expect(response).to have_http_status(400)
+      expect { hash_body = json }.to_not raise_exception
+      expect(hash_body).to match({})
+    end
+
+    it 'When invalid request, missing attributes values' do
+      hash_body = nil
+      invalid_request_body = @request_body.deep_dup
+      invalid_request_body[:data][:attributes].delete(:data)
+      post api_v1_team_inventory_column_list_items_path(
+        team_id: @teams.first.id,
+        inventory_id: @valid_inventory.id,
+        column_id: @list_column
+      ), params: invalid_request_body.to_json, headers: @valid_headers
+      expect(response).to have_http_status(400)
+      expect { hash_body = json }.to_not raise_exception
+      expect(hash_body).to match({})
+    end
+  end
+
+  describe 'PUT inventory_list_item, #update' do
+    before :all do
+      @valid_headers['Content-Type'] = 'application/vnd.api+json'
+      @request_body = {
+        data: {
+          id: @list_column.repository_list_items.first.id,
+          type: 'inventory_list_items',
+          attributes: { data: 'Updated' }
+        }
+      }
+    end
+
+    it 'Response with correct inventory list item' do
+      hash_body = nil
+      put api_v1_team_inventory_column_list_item_path(
+        id: @list_column.repository_list_items.first.id,
+        team_id: @teams.first.id,
+        inventory_id: @valid_inventory.id,
+        column_id: @list_column
+      ), params: @request_body.to_json, headers: @valid_headers
+      expect(response).to have_http_status 200
+      expect { hash_body = json }.not_to raise_exception
+      expect(hash_body[:data]).to match(
+        ActiveModelSerializers::SerializableResource
+          .new(@list_column.repository_list_items.first,
+               serializer: Api::V1::InventoryListItemSerializer)
+          .as_json[:data]
+      )
+      expect(@list_column.repository_list_items.first.data).to match('Updated')
+    end
+
+    it 'When invalid request, incorrect type' do
+      invalid_request_body = @request_body.deep_dup
+      invalid_request_body[:data][:type] = 'repository_rows'
+      hash_body = nil
+      put api_v1_team_inventory_column_list_item_path(
+        id: @list_column.repository_list_items.first.id,
+        team_id: @teams.first.id,
+        inventory_id: @valid_inventory.id,
+        column_id: @list_column
+      ), params: invalid_request_body.to_json, headers: @valid_headers
+      expect(response).to have_http_status(400)
+      expect { hash_body = json }.to_not raise_exception
+      expect(hash_body).to match({})
+    end
+
+    it 'When invalid request, missing attributes values' do
```

Comment:
I think it'd be also good to write a test to check with an invalid `id` parameter here, for `PUT`.